src/lib/core/evio.c

```diff
@@ -208,17 +208,21 @@ static int
 evio_service_entry_reuse_addr(const char *uri)
 {
 	struct uri u;
-	if (uri_parse(&u, uri) || u.service == NULL) {
+	if (uri_create(&u, uri) || u.service == NULL) {
+		uri_destroy(&u);
 		diag_set(IllegalParams, "invalid uri for bind: %s", uri);
 		return -1;
 	}
-	if (strncmp(u.host, URI_HOST_UNIX, u.host_len) != 0)
+	if (u.host != NULL && strcmp(u.host, URI_HOST_UNIX) != 0) {
+		uri_destroy(&u);
 		return 0;
+	}
 
-	struct sockaddr_un un;
-	snprintf(un.sun_path, sizeof(un.sun_path), "%.*s",
-		 (int)u.service_len, u.service);
+	struct sockaddr_un un = { 0 };
+	if (u.service != NULL)
```

Comment:
Nit: `u.service` can't be NULL here - you check it right above. Same comment for `evio_service_entry_bind`.

Reply:
fixed